### PR TITLE
FIX: 'discourse watch all' cmd posting topics of categories which are private

### PR DESCRIPTION
**Pull request by vinothkannans**
'/discourse watch all' command posting topics of categories which are private

https://meta.discourse.org/t/slack-official-plugin-sending-topics-from-staff-categories/49578

### CODE REVIEW

**Comment by mcwumbly** [comment]
I'm not familiar enough with the code these days to review this without
testing it.

Is it possible to add a test that fails before this fix to demonstrate this
fixes the issue?

File: plugin.rb [hunk 1]
```diff
@@ -185,7 +185,7 @@ def find_topic(topic_id, post_number)
   end
 
   DiscourseEvent.on(:post_created) do |post|
-    if SiteSetting.slack_enabled?
+    if SiteSetting.slack_enabled? && DiscourseSlack::Slack.guardian.can_see?(post)
```
Comment (tgxworld) [inline plugin.rb]: I'll move the logic into 

https://github.com/discourse/discourse-slack-official/blob/97ceba2ebda7f608ea9d6e9c8d83451ffc431589/lib/discourse_slack/slack.rb#L156-L157

**Comment by vinothkannans** [comment]
@mcwumbly I will do once the logic moved to above mentioned place.

**Comment by tgxworld** [comment]
Can we add a test case as well?

**Comment by vinothkannans** [comment]
Sure. I will do @tgxworld  :+1:

File: spec/jobs/notify_slack_spec.rb [hunk 2]
```diff
@@ -39,6 +39,20 @@
 
       expect(response[0]).to eq("success")
     end
+
+    it "should notify if user have permission to see the post" do
+      topic
+      DiscourseSlack::Slack.set_filter_by_id(category.id, "#general", "follow")
+      Guardian.any_instance.expects(:can_see?).with(post).returns(false)
```
Comment (tgxworld) [inline spec/jobs/notify_slack_spec.rb]: Let's not use a stub here. I prefer to just use an actual user object to test it. Stubs are fragile and isn't testing the underlying code properly.

**Review by tgxworld — CHANGES_REQUESTED** [review]

**Comment by tgxworld** [comment]
@vinothkannans Changes look good to me aside from a minor comment. Please squash your commits as well. :+1:

**Comment by vinothkannans** [comment]
it's done. now it can be merged

File: spec/jobs/notify_slack_spec.rb [hunk 3]
```diff
@@ -39,6 +39,25 @@
...
+      user = Fabricate(:user)
+
+      SiteSetting.slack_discourse_username = user.username
+      DiscourseSlack::Slack.set_filter_by_id(nil, "#general", "follow")
+
+      response = Jobs::NotifySlack.new.execute(post_id: post.id)
+      expect(response).to eq(nil)
+
+      category.read_restricted = false
+      category.save!
```
Comment (tgxworld) [inline spec/jobs/notify_slack_spec.rb:56]: Just a note, in the future you can just use `category.update!(read_restricted: false)`

**Comment by tgxworld** [comment]
Thank you @vinothkannans :+1:

**Pull request merged**